### PR TITLE
tests: llext: cleanup `testcase.yaml`

### LINKED ISSUE #80170: Architecture Limitations for llext on nRF9160-DK
**Describe the bug**

We are using the nRF9160-DK (ARM platform) and are running the llext examples. However, we encounter a NULL return when calling `llext_find_sym()`. Could you please confirm if there are any architecture limitations or known issues related to using llext on the nRF9160-DK?

We are using the module below to verify the llext functionality:
https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/subsys/llext/modules

Here are the configurations we tried:

We commented the `printk` statement, and a crash is observed. Logs are provided in the link below:
Crash logs: https://drive.google.com/file/d/1TWCxfk2OyxBhDFBD4J0x43b8Su7x49Sn/view?usp=sharing

Code reference: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/llext/modules/src/hello_world_ext.c#L24

Thank you for your support!

**To Reproduce**
1. Clone the repository and navigate to `samples/subsys/llext/modules`
2. Modify the configuration as described above
3. Run the project on the nRF9160-DK
4. Observe the crash and the NULL return from `llext_find_sym()`

**Expected behavior**
`llext_find_sym()` should not return NULL when using llext on the nRF9160-DK, and the example should run without crashing.

**Impact**
This issue is a showstopper as it prevents us from verifying the llext functionality on the nRF9160-DK.

**Logs and console output**
Crash logs: https://drive.google.com/file/d/1TWCxfk2OyxBhDFBD4J0x43b8Su7x49Sn/view?usp=sharing
Logs https://drive.google.com/file/d/1TWCxfk2OyxBhDFBD4J0x43b8Su7x49Sn/view?usp=drive_link

**Environment (please complete the following information):**
 - OS: [Zephyr OS 3.6]
 - Toolchain: [e.g., Zephyr SDK, ARM GCC]
 - Commit SHA or Version: [Please provide the commit SHA or version of the Zephyr project you are using]

### PR DESCRIPTION
**Pull request by pillo79**
A number of cleanups / improvements on the test descriptions and coverage:
- Remove the `readonly_fs_loader` case, as it is an exact duplicate of `readonly`. The FS test already runs where applicable, with per-board configs. 
- Remove old limitations (`S32ZE` already added to `platform_excluded`, issues with Apollo HAL have been fixed)
- Move the list of arch-specific MMU/MPU enables to a common config file and include it where necessary (also improving the situation with respect to #80170 and #80613).
- Fix the filter for `readonly_mmu` to actually let `arm` and `riscv` platforms through.

### CODE REVIEW

**Review by teburd — APPROVED** [review]

**Review by awojasinski — APPROVED** [review]

File: tests/subsys/llext/simple/testcase.yaml [hunk 1]
```diff
@@ -22,6 +20,8 @@ common:
     - mps2/an521/cpu0         # ARM Cortex-M33 (ARMv8-M ISA)
   extra_configs:
     - arch:arm64:CONFIG_LLEXT_HEAP_SIZE=128
+  extra_conf_files:
+    - prj.conf
```
[the remark this answers is not in the document]
Comment (pillo79) [reply tests/subsys/llext/simple/testcase.yaml:24]: Thought so too, but for some reason on my PC `prj.conf` was not being picked up without those lines. I didn't investigate further. Do you have any hunch how that could happen?

[the remark this answers is not in the document]
Comment (awojasinski) [reply tests/subsys/llext/simple/testcase.yaml:24]: I'll try to analyze what's going on. Until some results let's leave it as it is and merge it

**Review by mathieuchopstm — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@pillo79 we are only allowing bug fixes to be merged into the 4.0 release. Is this fixing an issue, if so kindly provide an associated issue linked to this PR.

**Comment by pillo79** [comment]
> @pillo79 we are only allowing bug fixes to be merged into the 4.0 release. Is this fixing an issue, if so kindly provide an associated issue linked to this PR.

Sorry I wasn't explicit. This PR _plus_ PR #80613 together fix issue #80170, where an user complained `west build` was not working as expected on Twister tests on hardware. 
They are independently addressing separate areas of the issue:
- #80613 will warn the user about  Twister options that are unsupported by `west build`
- this PR removes most of those options from LLEXT tests, so the users will not notice the issue.

**Pull request merged**